### ISSUE TITLE
Problem with recent Fedora updates

### ISSUE THREAD

**Issue by damonlynch** [sbraz/pymediainfo#43]
Fedora now has version 17.10 of libmediainfo. It appears as if pymediainfo 2.2.0 is incompatible with it -- or is there some step I must now take to update the version I have installed with pip? 

Thank you for your work! (I'm the author of Rapid Photo Downloader, which uses your wrapper).

**Comment by sbraz** [comment]
Hi, I'm not sure what your problem is. Version 2.2.0 is compatible with MediaInfo 17.10. If Fedora doesn't provide pymediainfo 2.2.0, you need to install it another way, indeed.

**Comment by damonlynch** [comment]
I'm not sure what the problem is either. But I'm getting reports from multiple users and I see it myself. You can see it in a Fedora VM with updates installed. It start happening about a week or two ago, when libmediainfo was updated. 

Calling this throws the OSError exception:
`pymediainfo.MediaInfo.parse('/dev/null')`

**Comment by sbraz** [comment]
Can you show me the whole traceback and the output from `pymediainfo.__version__`?

**Comment by sbraz** [comment]
I've been looking at your code. You should call `pymediainfo.MediaInfo.can_parse()` instead of trying to parse /dev/null.

**Comment by damonlynch** [comment]
Thanks for taking the time to investigate my code. I'll make the change you suggest. Meanwhile here is the output you requested:
```
>>> import pymediainfo
>>> pymediainfo.__version__
'2.2.0'
>>> pymediainfo.MediaInfo.parse('/dev/null')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/damon/.local/lib/python3.6/site-packages/pymediainfo/__init__.py", line 176, in parse
    lib = cls._get_library(library_file)
  File "/home/damon/.local/lib/python3.6/site-packages/pymediainfo/__init__.py", line 149, in _get_library
    return CDLL("libmediainfo.so.0")
  File "/usr/lib64/python3.6/ctypes/__init__.py", line 343, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: libmediainfo.so.0: cannot open shared object file: No such file or directory
```

**Comment by sbraz** [comment]
That's really surprising, this seems to indicate that libmediainfo is installed in a non-standard location. Do you know where the library is located? If you install the mediainfo binary, is it able to find its library? On my system, I have this:
```
$ ldd /usr/bin/mediainfo | grep libmediainfo.so.0 
        libmediainfo.so.0 => /usr/lib64/libmediainfo.so.0 (0x00007fa71d8dd000)
```

**Comment by sbraz** [comment]
OK I installed Fedora, for some reason the library is named `/usr/lib64/libmediainfo.so.17`… I'll see what I can do.

**Comment by sbraz** [comment]
```
# dnf repoquery -l libmediainfo |grep libmediainfo.so
Last metadata expiration check: 0:04:35 ago on Thu Nov 30 19:40:54 2017.
/usr/lib64/libmediainfo.so.0
/usr/lib64/libmediainfo.so.0.7.99
/usr/lib/libmediainfo.so.17
/usr/lib/libmediainfo.so.17.10
/usr/lib64/libmediainfo.so.17
/usr/lib64/libmediainfo.so.17.10
/usr/lib/libmediainfo.so.0
/usr/lib/libmediainfo.so.0.7.99
```
And yet `ls: cannot access '/usr/lib64/libmediainfo.so.0': No such file or directory`. Perhaps that's a bug with the Fedora package?

**Comment by damonlynch** [comment]
It sure looks like it. Would you prefer to file a bug in Fedora, or shall I do it?

**Comment by sbraz** [comment]
Looks like the problem lies with the upstream build system. I opened an upstream issue: https://github.com/MediaArea/MediaInfoLib/issues/695
I will only handle the new file name if this is indeed the expected behaviour. In the meantime, you could work around the issue by passing the `library_file` argument to `can_parse` and `parse` if the initial call fails.

**Comment by damonlynch** [comment]
Thanks @sbraz , I made the suggested change in my code. Good job finding the root cause of the problem!

**Comment by Vascom** [comment]
If upstream revert .so version I rebuld Fedora packages.
Also I can add pymediainfo to upstream if you check it, for example http://koji.russianfedora.pro/koji/taskinfo?taskID=59843

**Comment by sbraz** [comment]
@damonlynch hopefully you won't need the fix for a long time, upstream will soon fix the .so name.
@Vascom you mean add it to Fedora? Yeah it would be nice, what should I check exactly? If the tests run, it means the package is working :)

**Comment by Vascom** [comment]
@sbraz Yes, in Fedora official repo. Tests are not run because it has errors. I will create new issue with it and we try solve problem.

**Comment by Vascom** [comment]
pymedianfo added to Fedora repo. You can install package python2-pymediainfo or python3-pymediainfo.
It available now in rawhide and after about 3 days in testing for all other repos.
Will be available after about one week for Fedora 27, 26, 25 stable repo.
Will be available after about two weeks in EPEL (for RHEL/CentOS 6 and 7) stable repo.

Please report about all packaging bugs in RedHat bagtracker https://bugzilla.redhat.com

**Comment by sbraz** [comment]
@Vascom Thanks!
@damonlynch Were you able to test it? Can we close this issue?

**Comment by damonlynch** [comment]
@sbraz yes I was able to implement a solution, thank you. Users are now reporting it works. One suggestion / question I have is if it is possible for the documentation to note the version in which a feature was introduced? I ask because I realized just before I made a release of my project that the `library_file` argument works only with very recent releases of pymediainfo.  If so, I can open a new issue report regarding the documentation.
